cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/linkers/CassandraTestEventsMessagesLinker.java

```diff
@@ -94,6 +95,44 @@ public CassandraTestEventsMessagesLinker(TestEventMessagesOperator testEventsOpe
 			});
 	}
 	
+	@Override
+	public Collection<ExtendedTestEventId> getTestEventIds(StoredMessageId messageId) throws IOException
+	{
+		try
+		{
+			return getTestEventIdsAsync(messageId).get();
+		}
+		catch (Exception e)
+		{
+			throw new IOException("Error while getting IDs of test events linked to message "+messageId, e);
+		}
+	}
+	
+	@Override
+	public CompletableFuture<Collection<ExtendedTestEventId>> getTestEventIdsAsync(StoredMessageId messageId)
+	{
+		CompletableFuture<MappedAsyncPagingIterable<MessageTestEventEntity>> future = new AsyncOperator<MappedAsyncPagingIterable<MessageTestEventEntity>>(semaphore)
+				.getFuture(() -> messagesOperator.getTestEvents(instanceId, messageId.toString(), readAttrs));
+		
+		return future.thenCompose((rs) -> {

```

Comment:
We don't have any additional task to run so why don't use the `thenApply` method instead?

Reply:
Just because it was my mistake. Thank you for pointing this out.